### ISSUE TITLE
False positive for `dict-keys-not-iterating` w/ `--py3k`

### ISSUE THREAD

**Issue by asottile** [PyCQA/pylint#2137]
### Steps to reproduce

```python
max({1: True, 2: False}.keys())
```

### Current behavior

```
$ pylint --py3k  test.py
No config file found, using default configuration
************* Module test
W:  1, 4: dict.keys referenced when not iterating (dict-keys-not-iterating)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Expected behavior

No error

### pylint --version output

```
$ pylint --version
No config file found, using default configuration
pylint 1.9.1, 
astroid 1.6.4
Python 3.6.5 (default, Mar 30 2018, 06:41:53) 
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
```
